robot/basestation/static/js/roslibjs/ros-helpers.js

```diff
@@ -179,92 +171,107 @@ function initRosWeb () {
   battery_voltage_listener = new ROSLIB.Topic({
     ros: ros,
     name: 'battery_voltage',
-    messageType: 'std_msgs/Float32'
+    messageType: 'mcu_control/BatteryVoltage'
   })
   battery_voltage_listener.subscribe(function (message) {
-    // sets voltage to two decimal points
-    let voltage = message.data.toFixed(2)
-    $('#battery-voltage').text(voltage)
-
-    // if statement to control voltage indicator switching between acceptable(white) and unacceptable(red)
-    if ((voltage > MAX_VOLTAGE || voltage < MIN_VOLTAGE)) {
-      textColor('#battery-voltage', 'red')
-
-      if ($('#battery-voltage').attr('acceptable') === '1' && ALERT_ENABLE) {
-        $('#battery-voltage').attr('acceptable', '0')
-        errorSound()
-        if (voltage > MAX_VOLTAGE) {
-          navModalMessage('Warning: Voltage too high', 'Disconnect Battery first and then the BMS, and then discharge the Battery to 16.8V')
-        } else if (voltage < MIN_VOLTAGE){
-          navModalMessage('Warning: Voltage too low', 'Turn rover off, disconnect Battery and BMS, and then charge battery to 16.8V')
-        }
-      }
-    } else if (voltage > MAX_VOLTAGE - VOLTAGE_LEEWAY || voltage < MIN_VOLTAGE + VOLTAGE_LEEWAY){
-      if ($('#battery-voltage').attr('acceptable') === '0') {
+    if (message.gotVoltage) {
+      // sets voltage to two decimal points
+      let voltage = message.vbat.toFixed(2)
+      $('#battery-voltage').text(voltage)
+      $('#battery-voltage').siblings().show()
+
+      // if statement to control voltage indicator switching between acceptable(white) and unacceptable(red)
+      if ((voltage > MAX_VOLTAGE || voltage < MIN_VOLTAGE)) {
         textColor('#battery-voltage', 'red')
-      }
 
-    } else { 
-      if (voltage < MIN_VOLTAGE + VOLTAGE_WARNING) {
-        textColor('#battery-voltage', 'orange')
+        if ($('#battery-voltage').attr('acceptable') === '1' && ALERT_ENABLE) {
```

Comment:
You should make a method that wraps this jquery call. 
something like

`if(isVoltageAcceptable() && ALERT_ENABLE)`

so that if the business logic changes, aka if the way we are determining if the voltage is acceptable changes then we only need to change the contents of the methods.

Reply:
I don't really get this though because If I do this then shouldent I do the same for the temperatures?  But the temperature checks work using the $obj.attr.  So then I would just be creating a function that takes in the object to check which is kinda retarded imo.  If you still want it ill do it in the morning.  @MartensCedric